src/category_theory/morphism_property.lean

```diff
@@ -172,6 +172,53 @@ end
 
 end naturality_property
 
+section diagonal
+
+variables [has_pullbacks C] {P : morphism_property C}
+
+/-- For `P : morphism_property C`, `P.diagonal` is a morphism property that holds for `f : X ⟶ Y`
+whenever `P` holds for `X ⟶ Y xₓ Y`. -/
+def diagonal (P : morphism_property C) : morphism_property C :=
+λ X Y f, P (pullback.diagonal f)
+
+@[simp]
```

Comment:
Should this be a global simp lemma? Or just locally while setting up the API?
It feels like having this `simp` means that `P.diagonal` can't have its own API.